apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.DeleteObject.cs

```diff
@@ -64,6 +64,8 @@ private ObjectsResource.DeleteRequest CreateDeleteObjectRequest(string bucket, s
             GaxPreconditions.CheckNotNull(name, nameof(name));
             var request = Service.Objects.Delete(bucket, name);
             options?.ModifyRequest(request);
+            RetryOptions retryOptions = options?.RetryOptions ?? ((options?.IfGenerationMatch is null && options?.Generation is null) ? RetryOptions.Never : RetryOptions.IdempotentRetryOptions);
```

Comment:
I don't think we need two brackets here. Just one after ?? should be enough.
Also, I think we can format it to be more readable, something like 
```
 RetryOptions retryOptions = options?.RetryOptions ?? (options?.IfGenerationMatch is null && options?.Generation is null
                ? RetryOptions.Never
                : RetryOptions.IdempotentRetryOptions);
```

Please format as per editorconfig (My illustration may not be correctly indented)